bitcoinlib/services/groestlcoind.py

```diff
@@ -0,0 +1,271 @@
+            for location in config_locations:
+                cfn = Path(location, config_fn).expanduser()
+                if cfn.exists():
+                    break
+        else:
+            cfn = Path(BCL_DATA_DIR, 'config', configfile)
+        if not cfn or not cfn.is_file():
+            raise ConfigError("Config file %s not found. Please install bitcoin client and specify a path to config "
+                              "file if path is not default. Or place a config file in .bitcoinlib/bitcoin.conf to "
+                              "reference to an external server." % cfn)
+
+        try:
+            config.read(cfn)
+        except Exception:
+            with cfn.open() as f:
+                config_string = '[rpc]\n' + f.read()
+            config.read_string(config_string)
+
+        testnet = _read_from_config(config, 'rpc', 'testnet')
+        if testnet:
+            network = 'testnet'
+        if _read_from_config(config, 'rpc', 'rpcpassword') == 'specify_rpc_password':
+            raise ConfigError("Please update config settings in %s" % cfn)
+        if network == 'testnet':
+            port = 18332
+        else:
+            port = 8332
+        port = _read_from_config(config, 'rpc', 'rpcport', port)
+        server = '127.0.0.1'
+        server = _read_from_config(config, 'rpc', 'rpcconnect', server)
+        server = _read_from_config(config, 'rpc', 'bind', server)
+        server = _read_from_config(config, 'rpc', 'externalip', server)
+
+        url = "http://%s:%s@%s:%s" % (config.get('rpc', 'rpcuser'), config.get('rpc', 'rpcpassword'), server, port)
+        return GroestlcoindClient(network, url)
+
+    def __init__(self, network='bitcoin', base_url='', denominator=100000000, *args):
+        """
+        Open connection to bitcoin node
+
+        :param network: Bitcoin mainnet or testnet. Default is bitcoin mainnet
+        :type: str
+        :param base_url: Connection URL in format http(s)://user:password@host:port.
+        :type: str
+        :param denominator: Denominator for this currency. Should be always 100000000 (satoshis) for bitcoin
+        :type: str
+        """
+        if isinstance(network, Network):
+            network = network.name
+        if not base_url:
+            bdc = self.from_config('', network)
+            base_url = bdc.base_url
+            network = bdc.network
+        if len(base_url.split(':')) != 4:
+            raise ConfigError("Groestlcoind connection URL must be of format 'http(s)://user:password@host:port,"
+                              "current format is %s. Please set url in providers.json file or check bitcoin config "
+                              "file" % base_url)
+        if 'password' in base_url:
+            raise ConfigError("Invalid password in groestlcoind provider settings. "
+                              "Please replace default password and set url in providers.json or bitcoin.conf file")
```

Comment:
"Please replace default password and set url in providers.json or groestlcoin.conf file")